src/actions/actions/services/dispatchers/verusid/verusid.js

```diff
@@ -56,3 +65,193 @@ export const unlinkVerusId = async (iAddress, chain) => {
+        // If the request is older than 10 minutes, check info endpoint to see if it was accepted or rejected
+        let errorFound = false;
+        try {
+          if (pendingIds[ticker][iaddress].infoUri) {
+            const response = await axios.post(pendingIds[ticker][iaddress].infoUri);
+
+            if (response.data.result.state === primitives.LOGIN_CONSENT_PROVISIONING_RESULT_STATE_FAILED.vdxfid) {
+
+              const newDeepLinkNotification = new DeeplinkNotification (
+                "Retry",
+                [`${pendingIds[ticker][iaddress].provisioningName.split(".")[0]}@`, ` failed to create identity.`],
+                null,
+                null,
+                pendingIds[ticker][iaddress].loginRequest,
+                state.authentication.activeAccount.accountHash
+              ); 
+              await deleteProvisionedIds(iaddress, ticker);
+              await updatePendingVerusIds();
+              newDeepLinkNotification.icon = NOTIFICATION_ICON_ERROR;
+              newDeepLinkNotification.uid = pendingIds[ticker][iaddress].notificationUid;
+              await dispatchAddNotification(newDeepLinkNotification);
+              continue;
+            }
+          } 
+        } catch (e) {
+
+          if ((pendingIds[ticker][iaddress].createdAt + 1200) < Math.floor(Date.now() / 1000) &&
+                pendingIds[ticker][iaddress].status !== NOTIFICATION_TYPE_VERUSID_ERROR) {
+
+            pendingIds[ticker][iaddress].status = NOTIFICATION_TYPE_VERUSID_ERROR;
+            pendingIds[ticker][iaddress].error_desc = [`${pendingIds[ticker][iaddress].provisioningName}@`, `server connection error.`]
+            pendingIds[ticker][iaddress].createdAt = Math.floor(Date.now() / 1000) + 1200;
+            await setRequestedVerusId(iaddress, pendingIds[ticker][iaddress], ticker);
+            await updatePendingVerusIds();
+            errorFound = true;
+          }
+        } 
+
+        if (errorFound) {
+          const newDeepLinkNotification = new BasicNotification (
+            "",
+            pendingIds[ticker][iaddress].error_desc,
+            null,
+            state.authentication.activeAccount.accountHash
+          );  
+          newDeepLinkNotification.icon = NOTIFICATION_ICON_ERROR;
+          newDeepLinkNotification.uid = pendingIds[ticker][iaddress].notificationUid;
+          await dispatchAddNotification(newDeepLinkNotification);
+          continue;
+        }
+        
+      }
+
+      const identity = await getIdentity(system.system_id, iaddress);
+
+      if (identity.result &&
+        identity.result.identity.primaryaddresses
+          .indexOf(state.authentication.activeAccount.keys[ticker].vrpc.addresses[0]) > -1) {
+
+        pendingIds[ticker][iaddress].status = NOTIFICATION_TYPE_VERUSID_READY;
+
+        await setRequestedVerusId(iaddress, pendingIds[ticker][iaddress], ticker);
+        await linkVerusId(iaddress, identity.result.fullyqualifiedname, ticker);
```

Comment:
After this is linked, the user should be made aware if the linked ID has a different primary address included in it than the one in their wallet, and/or if the revocation and recovery IDs are not set to itself.